domain/application/errors/errors.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package errors
+
+import (
+	"github.com/juju/errors"
+)
+
+const (
+	// NotFound describes an error that occurs when the application being operated on
+	// does not exist.
+	NotFound = errors.ConstError("application not found")
```

Comment:
I'm unsure we want this, why not just use domain https://github.com/juju/juju/blob/main/domain/errors.go#L19

Reply:
This is very deliberate. We absolutely must move away from "generic" not found errors and use proper, bespoke domain specific ones. We have specific errors already for machine not found, model not found etc.